### PR TITLE
Compute `$isConstructor` in `ReflectionMethod` constructor

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/ReflectionMethod.php [hunk 1]
```diff
@@ -336,16 +339,24 @@ public function isStatic(): bool
...
         if ($declaringClass->inNamespace()) {
             return false;
         }
 
-        return strtolower($this->getName()) === strtolower($declaringClass->getShortName());
+        if ($declaringClass->isAnonymous()) {
+            return false;
+        }
+
+        return strtolower($name) === strtolower($declaringClass->getShortName());
```
Comment (Ocramius) [inline src/Reflection/ReflectionMethod.php:359]: I don't think this applies anymore: https://3v4l.org/sGLco

Comment (kukulich) [reply src/Reflection/ReflectionMethod.php:359]: There's was one revert already :) https://github.com/Roave/BetterReflection/pull/907

https://github.com/Roave/BetterReflection/issues/904#issuecomment-986588835

@ondrejmirtes Is it still relevant?

Comment (ondrejmirtes) [reply src/Reflection/ReflectionMethod.php:359]: You're right, it last worked in PHP 7. (And it never worked in namespaces.)

Comment (Ocramius) [reply src/Reflection/ReflectionMethod.php:359]: Let's kill it: we don't support PHP 7.4 anymore.

**Pull request closed without merging**